beets/util/__init__.py

```diff
@@ -442,17 +450,29 @@ def samefile(p1, p2):
     return shutil._samefile(syspath(p1), syspath(p2))
 
 
-def remove(path, soft=True):
-    """Remove the file. If `soft`, then no error will be raised if the
-    file does not exist.
+def remove(path, soft=True, remove_command=config["remove_command"].as_str()):
```

Comment:
Is it guaranteed that the config has actually been read by the time the util module is imported?

Reply:
The Beets development docs seem to imply that this behavior can be relied upon.

Reply:
Ok :+1: Do you have a link here? I'd actually be interested in reading up on this, it doesn't look obvious at all: I would have thought that this might depend on import order and thus be rather fragile.

Reply:
From https://github.com/beetbox/beets/blob/master/docs/dev/plugins.rst#read-configuration-options:

> If you want to access configuration values _outside_ of your plugin's section, import the config object from the beets module. That is, just put `from beets import config` at the top of your plugin and access values from there.

While we're operating outside the context of a plugin here, it seems like this is intended to work even outside a plugin's realm.

Reply:
> While we're operating outside the context of a plugin here, it seems like this is intended to work even outside a plugin's realm.

I'm not convinced ;) The thing is, for plugins I can see why it works: Plugins are imported manually in `beets.ui._load_plugins`. At that point, the main beets modules have surely been imported and set up. However, inside of beets core, that might not be the case. In fact, in `beets.ui._configure`, an overlay configuration given through the `--config` option might be applied. `beets.ui` already imports `beets.util` before that, so values from the overlay configuration will never make it to the `remove` kwarg default. For the "normal" non-overlay config I'm not sure, I haven't looked into confuse in a while, so I can't see right now when that would be loaded.

Reply:
I agree that changing the interface to `use_trash=True` is an improvement, so I just pushed an updated commit that switches to that argument. The `remove()` function will use an alternate command if a `remove_command` value is found in the configuration, but now otherwise falls back to `trash`, thereby improving durability. I believe these changes constitute a reasonable compromise solution to the concerns raised here.